spec/Section 3 -- Type System.md

```diff
@@ -1520,6 +1520,28 @@ define arguments or contain references to interfaces and unions, neither of
 which is appropriate for use as an input argument. For this reason, input
 objects have a separate type in the system.
 
+**OneOf Input Objects**
+
+OneOf Input Objects are a special variant of Input Objects where the type system
+asserts that exactly one of the fields must be set and non-null, all others
+being omitted. This is useful for representing situations where an input may be
+one of many different options.
+
+When using the type system definition language, the `@oneOf` directive is used
+to indicate that an Input Object is a OneOf Input Object (and thus requires
+exactly one of its field be provided):
+
+```graphql
+input UserUniqueCondition @oneOf {
+  id: ID
+  username: String
+  organizationAndEmail: OrganizationAndEmailInput
+}
+```
+
+In schema introspection, the `__Type.isOneOf` field will return {true} for OneOf
```

Comment:
I feel like there is a inconsistency here with the fact that in other cases where a `__Type` field does not apply it is null and not true/false.  eg `possibleTypes` never applies to Scalsrs and hence its null so following that logic,  `isOneOf` never applies to Scalars so it should be null 

> In schema introspection, the `__Type.isOneOf` field will return {true} for OneOf
> Input Objects, and {false} for all other Input Objects.

And in the merged JS implementation we have

```
isOneOf: {
        type: GraphQLBoolean,
        resolve: (type) => {
          if (isInputObjectType(type)) {
            return type.isOneOf;
benjie marked this conversation as resolved.
          }
        },
      },
```

which I think will return null (or undefined maybe) if its not a OneOf Input Object.

Reply:
ps..  I dont think this is a show stopper - but it seems inconsistent to me to other `__Type` fields

Reply:
You may have misread? It says false for all other **input objects**. It’s null if not an input object (though it doesn’t explicitly state that here, it doesn’t need to and this makes space for oneOf to be applied to other types in future).

Reply:
Thank you for the clarification.